### DISCUSSION TITLE
Screen flickering on a by tick basis with certain program running

### DISCUSSION THREAD

**Discussion by coaber** [cc-tweaked/CC-Tweaked#1357] [Q&A]
I initially wanted to make a bug report about this but decided against it, since the issue is most likely due to the code. Unfortunately with my limited knowledge and the project's lack of updates, i cannot find any sort of fix to the issue presented in the title. I am running CC 1.101.2, with a old Draconic Evolution program that sends information for the draconic reactor. I managed to fix it up to work with 1.18.2's DE version thanks to the help of some people on DE's discord, but one thing i can't find any concrete documentation on is the constant screen flicker.
Here's my updated fork of the program itself: https://github.com/coaber/DEReactor
I've managed to find some 'potential' methods to fix it such as using a buffer or a redraw function, but to be quite honest, i have no clue how to go about doing that, so my last place was to ask here.

**Comment by SquidDev** [comment]
It looks like you've managed to get buffering working since posting this, so not sure there's much more I can suggest :).

**Reply by coaber** [reply]
i kind of figured it out, thanks to someone in the CC discord, however running the program with the patch just throws that error 
![image](https://user-images.githubusercontent.com/26202628/222697965-1da61592-f5a6-4bf2-87c7-06f7cf523173.png)

**Reply by SquidDev** [reply]
It looks like you removed the the clear function it's trying to use [in this commit](https://github.com/coaber/DEReactor/commit/3ab543a56f4e25d360915b863a7ecd090b758454) - if you add it back it should work.

**Reply by coaber** [reply]
yes, i ended up just re-pasting the whole chunk of code from another backup i had stored locally (which is the initial one you saw) and ran it just to make sure, and it still shows that same error upon startup.

**Reply by coaber** [reply]
Quick update, adding this code back fixed the issue, it all works as intended with absolutely no flicker whatsoever. Thanks for the help!
